File: iOS/LittleStoneBookkeeping/FeedbackViewController.m
#import "FeedbackViewController.h"

@interface FeedbackViewController () <UITextViewDelegate>

@property (nonatomic, strong) UILabel *titleLabel;
@property (nonatomic, strong) UITextView *feedbackTextView;
@property (nonatomic, strong) UIButton *submitButton;
@property (nonatomic, strong) UILabel *placeholderLabel;

// 双击检测
@property (nonatomic, assign) NSTimeInterval lastTapTime;
@property (nonatomic, assign) NSTimeInterval doubleTapInterval;
@property (nonatomic, assign) NSInteger tapCount;

@end

@implementation FeedbackViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.lastTapTime = 0;
    self.doubleTapInterval = 0.5;
    self.tapCount = 0;
    
    [self setupUI];
    [self setupGestures];
}

- (void)setupUI {
    self.view.backgroundColor = [UIColor systemBackgroundColor];
    
    // 标题
    self.titleLabel = [[UILabel alloc] init];
    self.titleLabel.text = @"意见反馈";
    self.titleLabel.font = [UIFont boldSystemFontOfSize:20];
    self.titleLabel.textAlignment = NSTextAlignmentCenter;
    
    // 反馈文本框
    self.feedbackTextView = [[UITextView alloc] init];
    self.feedbackTextView.layer.borderWidth = 1;
    self.feedbackTextView.layer.borderColor = [UIColor systemGray4Color].CGColor;
    self.feedbackTextView.layer.cornerRadius = 8;
    self.feedbackTextView.font = [UIFont systemFontOfSize:16];
    self.feedbackTextView.delegate = self;
    self.feedbackTextView.textContainerInset = UIEdgeInsetsMake(12, 8, 12, 8);
    
    // 占位符
    self.placeholderLabel = [[UILabel alloc] init];
    self.placeholderLabel.text = @"请输入您的宝贵意见...";
    self.placeholderLabel.textColor = [UIColor systemGray3Color];
    self.placeholderLabel.font = [UIFont systemFontOfSize:16];
    
    // 提交按钮
    self.submitButton = [UIButton buttonWithType:UIButtonTypeSystem];
    [self.submitButton setTitle:@"提交" forState:UIControlStateNormal];
    self.submitButton.titleLabel.font = [UIFont boldSystemFontOfSize:18];
    self.submitButton.backgroundColor = [UIColor systemBlueColor];
    [self.submitButton setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    self.submitButton.layer.cornerRadius = 8;
    
    // 布局
    UIStackView *stackView = [[UIStackView alloc] initWithArrangedSubviews:@[
        self.titleLabel,
        self.feedbackTextView,
        self.submitButton
    ]];
    
    stackView.axis = UILayoutConstraintAxisVertical;
    stackView.spacing = 20;
    stackView.translatesAutoresizingMaskIntoConstraints = NO;
    
    [self.view addSubview:stackView];
    [self.view addSubview:self.placeholderLabel];
    
    [NSLayoutConstraint activateConstraints:@[
        [stackView.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor constant:40],
        [stackView.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor constant:20],
        [stackView.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor constant:-20],
        
        [self.feedbackTextView.heightAnchor constraintEqualToConstant:120],
        [self.submitButton.heightAnchor constraintEqualToConstant:50],
        
        [self.placeholderLabel.topAnchor constraintEqualToAnchor:self.feedbackTextView.topAnchor constant:12],
        [self.placeholderLabel.leadingAnchor constraintEqualToAnchor:self.feedbackTextView.leadingAnchor constant:12]
    ]];
    
    // 添加关闭按钮
    UIBarButtonItem *closeButton = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemClose target:self action:@selector(close)];
    self.navigationItem.rightBarButtonItem = closeButton;
}

- (void)setupGestures {
    // 为提交按钮添加点击手势
    UITapGestureRecognizer *tapGesture = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(handleSubmitTap:)];
    [self.submitButton addGestureRecognizer:tapGesture];
    self.submitButton.userInteractionEnabled = YES;
}

- (void)handleSubmitTap:(UITapGestureRecognizer *)gesture {
    NSTimeInterval currentTime = [[NSDate date] timeIntervalSince1970];
    
    if (currentTime - self.lastTapTime < self.doubleTapInterval) {
        self.tapCount++;
        if (self.tapCount >= 2) {
            // 检测到双击
            [self handleDoubleTap];
            self.tapCount = 0;
            return;
        }
    } else {
        self.tapCount = 1;
    }
    
    self.lastTapTime = currentTime;
    
    // 延迟重置计数
    __weak typeof(self) weakSelf = self;
    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(self.doubleTapInterval * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
        if (weakSelf.tapCount == 1) {
            // 单次点击处理
            [weakSelf handleSingleTap];
        }
        weakSelf.tapCount = 0;
    });
}

- (void)handleSingleTap {
    // 单次点击：正常提交逻辑
    NSString *feedbackText = self.feedbackTextView.text;
    NSString *trimmedText = [feedbackText stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
    
    if (trimmedText.length == 0) {
        [self showAlertWithMessage:@"请输入反馈内容"];
        return;
    }
    
    // 模拟提交成功
    [self showAlertWithMessage:@"反馈提交成功！"];
    
    // 清空文本框
    self.feedbackTextView.text = @"";
    self.placeholderLabel.hidden = NO;
    
    // 关闭键盘
    [self.feedbackTextView resignFirstResponder];
}

- (void)handleDoubleTap {
    // 双击：切换界面逻辑（仅在文本框为空时生效）
    NSString *feedbackText = self.feedbackTextView.text;
    NSString *trimmedText = [feedbackText stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
    
    if (trimmedText.length > 0) {
        // 文本框有内容，不执行切换
        return;
    }
    
    // 直接执行界面切换
    [self performInterfaceSwitch];
}

- (void)performInterfaceSwitch {
    // 执行界面切换
    __weak typeof(self) weakSelf = self;
    [self dismissViewControllerAnimated:YES completion:^{
        if (weakSelf.interfaceSwitchHandler) {
            weakSelf.interfaceSwitchHandler();
        }
    }];
}

- (void)showAlertWithMessage:(NSString *)message {
    UIAlertController *alert = [UIAlertController alertControllerWithTitle:nil message:message preferredStyle:UIAlertControllerStyleAlert];
    [alert addAction:[UIAlertAction actionWithTitle:@"确定" style:UIAlertActionStyleDefault handler:nil]];
    [self presentViewController:alert animated:YES completion:nil];
}

- (void)close {
    [self dismissViewControllerAnimated:YES completion:nil];
}

#pragma mark - UITextViewDelegate

- (void)textViewDidChange:(UITextView *)textView {
    self.placeholderLabel.hidden = textView.text.length > 0;
}

- (void)textViewDidBeginEditing:(UITextView *)textView {
    self.placeholderLabel.hidden = YES;
}

- (void)textViewDidEndEditing:(UITextView *)textView {
    self.placeholderLabel.hidden = textView.text.length > 0;
}

@end

File: iOS/LittleStoneBookkeeping/FeedbackViewController.swift
import UIKit

class FeedbackViewController: UIViewController {
    
    var interfaceSwitchHandler: (() -> Void)?
    
    private let titleLabel = UILabel()
    private let feedbackTextView = UITextView()
    private let submitButton = UIButton(type: .system)
    private let placeholderLabel = UILabel()
    
    // 双击检测
    private var lastTapTime: TimeInterval = 0
    private let doubleTapInterval: TimeInterval = 0.5
    private var tapCount = 0
    
    override func viewDidLoad() {
        super.viewDidLoad()
        setupUI()
        setupGestures()
    }
    
    private func setupUI() {
        view.backgroundColor = .systemBackground
        
        // 标题
        titleLabel.text = "意见反馈"
        titleLabel.font = UIFont.boldSystemFont(ofSize: 20)
        titleLabel.textAlignment = .center
        
        // 反馈文本框
        feedbackTextView.layer.borderWidth = 1
        feedbackTextView.layer.borderColor = UIColor.systemGray4.cgColor
        feedbackTextView.layer.cornerRadius = 8
        feedbackTextView.font = UIFont.systemFont(ofSize: 16)
        feedbackTextView.delegate = self
        feedbackTextView.textContainerInset = UIEdgeInsets(top: 12, left: 8, bottom: 12, right: 8)
        
        // 占位符
        placeholderLabel.text = "请输入您的宝贵意见..."
        placeholderLabel.textColor = .systemGray3
        placeholderLabel.font = UIFont.systemFont(ofSize: 16)
        
        // 提交按钮
        submitButton.setTitle("提交", for: .normal)
        submitButton.titleLabel?.font = UIFont.boldSystemFont(ofSize: 18)
        submitButton.backgroundColor = .systemBlue
        submitButton.setTitleColor(.white, for: .normal)
        submitButton.layer.cornerRadius = 8
        
        // 布局
        let stackView = UIStackView(arrangedSubviews: [titleLabel, feedbackTextView, submitButton])
        stackView.axis = .vertical
        stackView.spacing = 20
        stackView.translatesAutoresizingMaskIntoConstraints = false
        
        view.addSubview(stackView)
        view.addSubview(placeholderLabel)
        
        NSLayoutConstraint.activate([
            stackView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 40),
            stackView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 20),
            stackView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -20),
            
            feedbackTextView.heightAnchor.constraint(equalToConstant: 120),
            submitButton.heightAnchor.constraint(equalToConstant: 50),
            
            placeholderLabel.topAnchor.constraint(equalTo: feedbackTextView.topAnchor, constant: 12),
            placeholderLabel.leadingAnchor.constraint(equalTo: feedbackTextView.leadingAnchor, constant: 12)
        ])
        
        // 添加关闭按钮
        let closeButton = UIBarButtonItem(barButtonSystemItem: .close, target: self, action: #selector(close))
        navigationItem.rightBarButtonItem = closeButton
    }
    
    private func setupGestures() {
        // 为提交按钮添加点击手势
        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(handleSubmitTap))
        submitButton.addGestureRecognizer(tapGesture)
        submitButton.isUserInteractionEnabled = true
    }
    
    @objc private func handleSubmitTap(_ gesture: UITapGestureRecognizer) {
        let currentTime = Date().timeIntervalSince1970
        
        if currentTime - lastTapTime < doubleTapInterval {
            tapCount += 1
            if tapCount >= 2 {
                // 检测到双击
                handleDoubleTap()
                tapCount = 0
                return
            }
        } else {
            tapCount = 1
        }
        
        lastTapTime = currentTime
        
        // 延迟重置计数
        DispatchQueue.main.asyncAfter(deadline: .now() + doubleTapInterval) { [weak self] in
            if self?.tapCount == 1 {
                // 单次点击处理
                self?.handleSingleTap()
            }
            self?.tapCount = 0
        }
    }
    
    private func handleSingleTap() {
        // 单次点击：正常提交逻辑
        guard let feedbackText = feedbackTextView.text, !feedbackText.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty else {
            showAlert(message: "请输入反馈内容")
            return
        }
        
        // 模拟提交成功
        showAlert(message: "反馈提交成功！")
        
        // 清空文本框
        feedbackTextView.text = ""
        placeholderLabel.isHidden = false
        
        // 关闭键盘
        feedbackTextView.resignFirstResponder()
    }
    
    private func handleDoubleTap() {
        // 双击：切换界面逻辑（仅在文本框为空时生效）
        guard feedbackTextView.text?.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty ?? true else {
            // 文本框有内容，不执行切换
            return
        }
        
        // 直接执行界面切换
        performInterfaceSwitch()
    }
    
    private func performInterfaceSwitch() {
        // 执行界面切换
        dismiss(animated: true) { [weak self] in
            self?.interfaceSwitchHandler?()
        }
    }
    
    private func showAlert(message: String) {
        let alert = UIAlertController(title: nil, message: message, preferredStyle: .alert)
        alert.addAction(UIAlertAction(title: "确定", style: .default))
        present(alert, animated: true)
    }
    
    @objc private func close() {
        dismiss(animated: true)
    }
}

extension FeedbackViewController: UITextViewDelegate {
    func textViewDidChange(_ textView: UITextView) {
        placeholderLabel.isHidden = !textView.text.isEmpty
    }
    
    func textViewDidBeginEditing(_ textView: UITextView) {
        placeholderLabel.isHidden = true
    }
    
    func textViewDidEndEditing(_ textView: UITextView) {
        placeholderLabel.isHidden = !textView.text.isEmpty
    }
}